### ISSUE TITLE
expose `config` and `key` to mapped methods

### ISSUE THREAD

**Issue by jonschlinkert** [doowb/map-config#11]
currently, there is no way to set or update a non-object value. but if `config` and `key` are exposed, the user could do something like:

``` js
app.config.map('foo', function(val, key, config, next) {
  if (!val) config[key] = 'bar';
  next();
});
```

**Comment by doowb** [comment]
Yeah, this sounds good.
I don't think `map-config` will re-process through, just allow a user to update if they want.

**Comment by doowb** [comment]
Published as 0.5.0 since this is a breaking change when using async methods.
